src/publication.rs

```diff
@@ -5,6 +5,55 @@ use crate::{
     ProtocolError, QoS, Retain,
 };
 
+pub trait ToPayload {
+    type Error;
+    fn serialize(&self, buffer: &mut [u8]) -> Result<usize, Self::Error>;
+}
+
+impl<'a> ToPayload for &'a [u8] {
+    type Error = ();
+
+    fn serialize(&self, buffer: &mut [u8]) -> Result<usize, Self::Error> {
+        if buffer.len() < self.len() {
+            return Err(());
+        }
+        buffer[..self.len()].copy_from_slice(self);
+        Ok(self.len())
+    }
+}
+
+impl<const N: usize> ToPayload for [u8; N] {
```

Comment:
Coersion doesn't appear to be working for the trait object because these are required for some of the unit tests:

Upon removal, various errors such as these are encountered:
```
note: required by a bound in `Pub`
   --> src\packets.rs:77:23
    |
77  | pub struct Pub<'a, P: crate::publication::ToPayload> {
    |                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ required by this bound in `Pub`

error[E0277]: the trait bound `&[{integer}; 2]: ToPayload` is not satisfied
   --> src\packets.rs:384:66
    |
384 |         let message = MqttSerializer::pub_to_buffer(&mut buffer, &publish).unwrap();
    |                       -----------------------------              ^^^^^^^^ the trait `ToPayload` is not implemented for `&[{integer}; 2]`
    |                       |
    |                       required by a bound introduced by this call
    |
    = help: the trait `ToPayload` is implemented for `&'a [u8]`
note: required by a bound in `MqttSerializer::<'a>::pub_to_buffer`
   --> src\ser\mod.rs:144:29
    |
144 |     pub fn pub_to_buffer<P: crate::publication::ToPayload>(
    |                             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ required by this bound in `MqttSerializer::<'a>::pub_to_buffer`

error[E0277]: the trait bound `&[{integer}; 2]: ToPayload` is not satisfied
   --> src\packets.rs:384:23
    |
384 |         let message = MqttSerializer::pub_to_buffer(&mut buffer, &publish).unwrap();
    |                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `ToPayload` is not implemented for `&[{integer}; 2]`
    |
    = help: the trait `ToPayload` is implemented for `&'a [u8]`

error[E0277]: the trait bound `&[{integer}; 2]: ToPayload` is not satisfied
   --> src\packets.rs:384:23
    |
384 |         let message = MqttSerializer::pub_to_buffer(&mut buffer, &publish).unwrap();
    |                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `ToPayload` is not implemented for `&[{integer}; 2]`
    |
    = help: the trait `ToPayload` is implemented for `&'a [u8]`

For more information about this error, try `rustc --explain E0277`.
error: could not compile `minimq` (lib test) due to 16 previous errors
warning: build failed, waiting for other jobs to finish...
```

Reply:
Ack!